### PR TITLE
chore: supporting missing auth config options

### PR DESCRIPTION
**Pull request by Jordan-Nelson**
*Description of changes:*
- add support for auth config options missing from Amplify Outputs schema

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by NikaHsn — APPROVED** [review]

File: packages/amplify_core/lib/src/config/amplify_outputs/auth/oauth_outputs.dart [hunk 1]
```diff
@@ -38,9 +43,39 @@ class OAuthOutputs
...
+  /// Optional query params in the sign in URI.
+  ///
+  /// Note: This attribute is not part of the Amplify Outputs schema.
+  /// This exists for backwards compatibility with the Gen 1 config.
+  @internal
+  final Map<String, String>? redirectSignOutUriQueryParameters;
+
+  /// The OAuth token URI.
+  ///
+  /// Defaults to '/oauth2/token'.
```
Comment (Equartey) [inline packages/amplify_core/lib/src/config/amplify_outputs/auth/oauth_outputs.dart:65]: Q: Where is the default set?

**Review by Equartey — APPROVED** [review]

**Review by tyllark — APPROVED** [review]

File: packages/amplify_core/lib/src/config/amplify_outputs/auth/oauth_outputs.dart (hunk 1 above)
Comment (Jordan-Nelson) [reply packages/amplify_core/lib/src/config/amplify_outputs/auth/oauth_outputs.dart:65]: In the auth plugin

File: packages/amplify_core/lib/src/config/amplify_outputs/auth/oauth_outputs.dart [hunk 2]
```diff
@@ -19,7 +20,11 @@ class OAuthOutputs
     required this.domain,
     required this.scopes,
     required this.redirectSignInUri,
+    this.redirectSignInUriQueryParameters,
```
Comment (NikaHsn) [inline packages/amplify_core/lib/src/config/amplify_outputs/auth/oauth_outputs.dart:23]: rename to `signInUriQueryParameters`

File: packages/amplify_core/lib/src/config/amplify_outputs/auth/oauth_outputs.dart [hunk 3]
```diff
@@ -19,7 +20,11 @@ class OAuthOutputs
...
     required this.redirectSignOutUri,
+    this.redirectSignOutUriQueryParameters,
```
Comment (NikaHsn) [inline packages/amplify_core/lib/src/config/amplify_outputs/auth/oauth_outputs.dart:25]: rename to `signOutUriQueryParameters`

File: packages/amplify_core/lib/src/config/amplify_outputs/auth/oauth_outputs.dart (hunk 2 above)
Comment (NikaHsn) [reply packages/amplify_core/lib/src/config/amplify_outputs/auth/oauth_outputs.dart:23]: we alos need to add optional `signInUri` field
`final String? signInUri;`

File: packages/amplify_core/lib/src/config/amplify_outputs/auth/oauth_outputs.dart (hunk 3 above)
Comment (NikaHsn) [reply packages/amplify_core/lib/src/config/amplify_outputs/auth/oauth_outputs.dart:25]: we alos need to add optional `signOutUri` field
`final String? signOutUri;`

**Pull request merged**